File: services/shop-backend/src/main/java/io/miragon/shop/adapter/outbound/persistence/article/ArticleMapper.java
package io.miragon.shop.adapter.outbound.persistence.article;

import io.miragon.shop.domain.article.Article;
import io.miragon.shop.domain.article.ArticleDescription;
import io.miragon.shop.domain.article.ArticleId;
import io.miragon.shop.domain.article.ArticleName;
import io.miragon.shop.domain.shared.Price;

public final class ArticleMapper {
    private ArticleMapper() {}

    public static Article toDomain(ArticleEntity e) {
        return new Article(
                new ArticleId(e.getId()),
                new ArticleName(e.getName()),
                new ArticleDescription(e.getDescription()),
                new Price(e.getPrice())
        );
    }

    public static ArticleEntity toEntity(Article a) {
        return new ArticleEntity(
                a.getId().getValue(),
                a.getName().getValue(),
                a.getDescription().getValue(),
                a.getPrice().getValue()
        );
    }
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/adapter/outbound/persistence/article/ArticleMapper.kt
package io.miragon.shop.adapter.outbound.persistence.article

import io.miragon.shop.domain.article.Article
import io.miragon.shop.domain.article.ArticleDescription
import io.miragon.shop.domain.article.ArticleId
import io.miragon.shop.domain.article.ArticleName
import io.miragon.shop.domain.shared.Price

object ArticleMapper {
    fun ArticleEntity.toDomain(): Article = Article(
        id = ArticleId(id),
        name = ArticleName(name),
        description = ArticleDescription(description),
        price = Price(price)
    )

    fun ArticleEntity.Companion.toEntity(
        article: Article
    ): ArticleEntity = ArticleEntity(
        id = article.id.value,
        name = article.name.value,
        description = article.description.value,
        price = article.price.value
    )
} 